First stab at add or update item. PHP code sample & intro text.

adetch opened the pull request.

danmcc commented on source/includes/_autocomplete_items.md:
```diff
@@ -82,6 +82,58 @@ image_url | No | A URL that points to an image you'd like displayed next to some
+
+### HTTP Request
+
+`POST https://ac.cnstrc.com/v1/item?force=1&autocomplete_key=[your autocomplete key]`
```
This should be PUT

adetch replied:
ah, of course

danmcc commented on source/includes/_autocomplete_items.md:
```diff
@@ -82,6 +82,58 @@ image_url | No | A URL that points to an image you'd like displayed next to some
+
+ > The above commands return a 204 Success response on success.
+
+An add or update operation (also known as an `UPSERT`) updates an item in your autocomplete index if it already exists, or adds the item to your index if it does not yet exist. 
```
One question devs will have at this point is, "how does Constructor.io identify whether an item exists?"  We should mention here that it's based on the combo of the item_name and the autocomplete_section OR the customer ID.  @EliFinkelshteyn -- is that right?

EliFinkelshteyn replied:
Yes. Customer ID if it exist, otherwise just name.

danmcc commented on source/includes/_autocomplete_items.md:
```diff
@@ -82,6 +82,58 @@ image_url | No | A URL that points to an image you'd like displayed next to some
+
+An add or update operation (also known as an `UPSERT`) updates an item in your autocomplete index if it already exists, or adds the item to your index if it does not yet exist. 
+
+To add or update an item in your autocomplete index, use the `POST /item` call, with `?force=1`. Options are the same as for the standard `Add an Item` call: `item_name` and `autocomplete_section` are required and all other optional parameters are supported.
```
"all other optional parameters are supported" reads a little funny to me.  I think just "all other parameters are optional" is a bit more standard.

adetch commented:
@danmcc do you want to review the wording here & examples for batch add?

adetch commented:
@danmcc I don't like the use of Search Suggestions for all the examples... most of our clients are adding Products and Categories to their autocomplete and in all such cases need to pass in additional information (url, image, etc). So I'd like to switch to Product-oriented examples moving forward.

I'm not sure, though that the extended examples I'm giving for PHP make sense. Thoughts? Let me know and I can apply whatever level of detail we settle on to the other language examples.

EliFinkelshteyn commented on source/includes/_autocomplete_items.md:
```diff
@@ -159,6 +213,59 @@ Parameter | Required? | Description
+  array("item_name" => "hammer", "id" => 3920, "url" => "/products/hammer", "image_url" => "/images/hammer.jpg"),
+  array("item_name" => "power drill", "id" => 3921, "url" => "/products/power-drill", "image_url" => "/images/power-drill.jpg")
+);
+
```
Did you test this? I'm not sure the first addOrUpdateBatch will work if `Products` requires a url. I don't think it should.

danmcc commented:
@adetch Using Products is fine, but as Eli pointed out, it will complicate all the examples by requiring us to add more metadata.  That's ok with me, but we'll just have to update everything.  Let's do that in a separate branch so we can release the upsert stuff asap.  If there are any other details to work out on this branch, let's just find each other on Slack and hammer them out there.

danmcc commented:
@adetch I added examples for Perl, Ruby, and Javascript.  I also simplified your PHP examples so we can get this out the door.  I like your added instructions, but it will take a little more time to think through Product examples, so let's do that separately.

Note we still don't have docs or code for the Python, Java, and C# libraries.  Let's discuss tomorrow how to proceed.

adetch commented:
> Let's do that in a separate branch so we can release the upsert stuff asap. 
:+1: 

> I added examples for Perl, Ruby, and Javascript. I also simplified your PHP examples so we can get this out the door. I like your added instructions, but it will take a little more time to think through Product examples, so let's do that separately.

awesome! & agreed
